src/plugins/factoids/factoidsPlugin.js

```diff
@@ -1,35 +1,66 @@
-const facts = require('./facts.json').factoids;
 const fs = require('fs');
+const { promisify } = require('util');
+const { getStore } = require('./storage.persistent');
+
+const readFile = promisify(fs.readFile);
+const writeFile = promisify(fs.writeFile);
+
+async function checkAndSaveDisabled(msg) {
+  const safeTo = String(msg.channel)
+    .replace(/[^a-zA-Z0-9#]/g, (char) => '-' + char.codePointAt(0).toString(16) + '-')
+    .replace(/#/g, '-H')
+    .replace(/^([^-])/, '-$1');
```

Comment:
This code will be removed eventually, but if there's a simpler way or more correct way to write it, let me know.

Reply:
This function is to create a safe filename right? why not using something like http://npm.im/slugify?

Also this file is to avoid that both jellobot and ecmabot runs factoids? is there a cleaner way to detect the presence of ecmabot user?

else you could rewrite `.replace(/^([^-])/, '-$1')` in `.replace(/^(?!-)/, '-')`

Reply:
Yeah, slugify would be fine. It's just going to consider `##foo` and `#foo` to be the same, but that really doesn't matter. Will change to that.

I think it would be too complicated to check ecmabot's presence with the channel list, and these disable operations persist across restarts of the bot, so I'm not super worried about it.